services/github-service-api/src/main/java/org/kontinuity/catapult/service/github/api/GitHubRepository.java

```diff
@@ -23,4 +24,18 @@
      * @return the github.com page for the repository
      */
     URI getHomepage();
+
+    /**
+	 * @param path
+	 *            the path to the file
+	 * @return the URL of the file with the given {@code} in the GitHub
+	 *         repository, <code>null</code> if no file with the given
+	 *         {@code path} exists in the repository.
+	 */
+	URI getDownloadUrl(String path);
```

Comment:
So then: `URI getDownloadUri(String path) throws InvalidPathException` or something similar where `InvalidPathException` extends `IllegalArgumentException` (and is thus unchecked) or you can make it checked if you feel that's more appropriate.